pkg/operator/collection.go

```diff
@@ -462,6 +469,41 @@ func (r *collectionReconciler) makeCollectorConfig(ctx context.Context, spec *mo
 		}
 	}
 
+	if err := r.client.List(ctx, &NodeMons); err != nil {
+		return nil, fmt.Errorf("failed to list NodeMonitorings: %w", err)
+	}
+	// The following job names are reserved by GMP for NodeMonitoring. They will not be generated if kubeletScraping is enabled.
+	var (
+		reservedCAdvisorJobName = "gmp-cadvisor-metrics"
+		reservedKubeletJobName  = "gmp-kubelet-metrics"
+	)
+	// Mark status updates in batch with single timestamp.
+	for _, nm := range NodeMons.Items {
+		if spec.KubeletScraping != nil && (nm.Name == reservedKubeletJobName || nm.Name == reservedCAdvisorJobName) {
+			logger.Info("NodeMonitoring job %s not applied because KubeletScraping is enabled.", nm.Name)
```

Comment:
```suggestion
			logger.Info("NodeMonitoring job %s not applied because OperatorConfig.collector.kubeletScraping is enabled.", nm.Name)
```

Perhaps to be precise?